hazelcast/test/src/compact/serialization/generic_record_factory.h

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "hazelcast/client/serialization/generic_record.h"
+
+#include "../../TestHelperFunctions.h"
+
+namespace hazelcast {
+namespace client {
+namespace test {
+namespace compact {
+
+inline serialization::generic_record::generic_record
+create_generic_record(const std::string& type_name = random_string())
+{
+    using namespace serialization::generic_record;
+
+    generic_record nested = generic_record_builder{ "nested_" + type_name }
+                              .set_boolean("nested_boolean", true)
+                              .build();
+    return generic_record_builder{ type_name }
+      .set_int32("id", 1)
```

Comment:
Will we support char16_t type for generic record?

Reply:
Do you mean field name or ?

Reply:
I realized char is not supported. So nothing to be fixed